sync-core/src/main/java/com/cloudant/sync/query/QueryExecutor.java

```diff
@@ -205,7 +227,9 @@ private boolean validateSortDocument(List<Map<String, String>> sortDocument) {
                 if (accumulator == null) {
                     accumulator = new HashSet<String>(childIds);
                 } else {
-                    accumulator = Sets.intersection(accumulator, childIds);
+                    if (childIds != null) {
+                        accumulator = Sets.intersection(accumulator, childIds);
+                    }
```

Comment:
Moved check for null to after childIds assignment.  https://github.com/cloudant/sync-android/commit/6179fea1b1139126f9e958e9eaf59fdea174709a#diff-80ec4d1d12847da53428d6c14a88dbb7R227